### PR TITLE
Matlab new driver fix ci

### PR DESCRIPTION
**Pull request by obejarano**
Tested all Matlab scripts on MEB and they all work.
TODO: continue to work on reliability in downlink (rl_dl_mimo_NEW)

### CODE REVIEW

File: MATLAB/rl_siso_NEW.m [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+        mimo_handle.mimo_update_sdr_param('rxgain', rxg_opt);
+    end
+
+    if strcmp(tx_direction, 'uplink')
+        [rx_vec_iris_tmp, numGoodFrames, ~] = mimo_handle.mimo_txrx_uplink(tx_vec_iris, N_FRM, N_ZPAD_PRE);
+
+    elseif strcmp(tx_direction, 'downlink')
+        [rx_vec_iris_tmp, numGoodFrames, ~] = mimo_handle.mimo_txrx_downlink(tx_vec_iris, N_FRM, N_ZPAD_PRE);
+
+    elseif strcmp(tx_direction, 'hub_sync')
```
Comment (rdoosty) [inline MATLAB/rl_siso_NEW.m:207]: `hub_sync` name is confusing. Perhaps something to indicate direction and the sync mode?

File: MATLAB/rl_siso_NEW.m [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+    if DEBUG
+        figure; plot(abs(rx_vec_iris));
+    end
+    %% Correlate for LTS
+    % Complex cross correlation of Rx waveform with time-domain LTS
+    a = 1;
+    unos = ones(size(preamble.'))';
+    v0 = filter(flipud(preamble'),a,rx_vec_iris);
+    v1 = filter(unos,a,abs(rx_vec_iris).^2);
+    m_filt = (abs(v0).^2)./v1; % normalized correlation
```
Comment (rdoosty) [inline MATLAB/rl_siso_NEW.m:246]: I have had issues with this match filtering scheme (consistently wrong offset). I am not sure if you have seen it. But we may want to do it differently.

File: MATLAB/mimo_driver.py [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+                #plt.plot(np.abs(tmptmp))
+
+        #plt.show()
+        [ue.unset_corr() for ue in self.ue_obj]
+        self.reset_frame()
+
+        return rx_data, good_frame_id, numRxSyms
+
+
+    def txrx_dl_sound(self, tx_data_mat_re, tx_data_mat_im, num_frames, nsamps_pad=160, max_try=30, python_mode=False):
```
Comment (rdoosty) [inline MATLAB/mimo_driver.py:340]: This function is very similar to the `txrx_downlink`? Can they be merged?

File: MATLAB/rl_siso_NEW.m (hunk 2 above)
Comment (obejarano) [reply MATLAB/rl_siso_NEW.m]: yeah, not a big deal with SISO but I'm doing it differently in other scripts. Need to check if I changed them all though

File: MATLAB/mimo_driver.py (hunk 3 above)
Comment (obejarano) [reply MATLAB/mimo_driver.py:340]: they are different but we could do that. I'll leave that as a TODO. don't want to break anything after all the testing. I'll add it to my next change

File: MATLAB/rl_siso_NEW.m (hunk 1 above)
Comment (obejarano) [reply MATLAB/rl_siso_NEW.m]: changed it

File: MATLAB/iris_py.py [hunk 4]
```diff
@@ -285,19 +318,36 @@ def recv_stream_tdd(self):
...
+                                print("BAD reading stream, index {}, : (ret:{})".format(ret, k))
+                                low_signal_a = True
+                                low_signal_b = True
+                                break
+
+                        print("reading stream: (ret:{}), {}, {}, {}".format(ret, ret.timeNs, ret.timeNs>>32, (ret.timeNs>>16)&0xFFFF))
+                        data_id = m*n_R+k
+                        rx_frames_a[data_id*in_len: (data_id*in_len + in_len)] = wave_rx_a
+                        rx_frames_b[data_id*in_len: (data_id*in_len + in_len)] = wave_rx_b
+                        if np.max(np.abs(wave_rx_a)) < 0.01:
```
[the remark this answers is not in the document]
Comment (obejarano) [reply MATLAB/iris_py.py:343]: might be too high but it has been working well. I'll check with lower values too

**Review by rdoosty — APPROVED** [review]

**Pull request merged**